configure.ac

```diff
@@ -851,25 +861,23 @@
     # TODO: see whether the code can be fixed to avoid -Wno-unused
     common_cflags="-O2 -fno-strict-aliasing -fwrapv -mms-bitfields"
     internal_cppflags='-D__USE_MINGW_ANSI_STDIO=0 -DUNICODE -D_UNICODE'
-    internal_cppflags="$internal_cppflags -DWINDOWS_UNICODE="
-    internal_cppflags="${internal_cppflags}\$(WINDOWS_UNICODE)"],
+    internal_cppflags="$internal_cppflags -DWINDOWS_UNICODE=$windows_unicode"],
```

Comment:
Do I understand correctly that the problem is in `ocamltest` getting the uninterpreted Makefile variable instead of the value?
An update to `README.win32.adoc:340` to state that it should really be changed in `configure`, maybe?

Reply:
> Do I understand correctly that the problem is in ocamltest getting the uninterpreted Makefile variable instead of the value?

No, it's getting an empty value, `-DWINDOWS_UNICODE=` is what's passed. It's not getting the verbatim text `$(WINDOWS_UNICODE)`. The same of `$(ROOTDIR)` when building FlexDLL.

> An update to [README.win32.adoc:340](https://github.com/ocaml/ocaml/blob/e719a1f78c825c1fff4c10c2c47e044149da4caf/README.win32.adoc?plain=1#L339-L342) to state that it should really be changed in `configure`, maybe?

The original text suggesting of tweaking `Makefile.config` seems fishy to me. The `configure` script is supposed to read `WINDOWS_UNICODE_MODE` from the environment. I'm not sure how everything works here.

Reply:
The documentation is outdated and predates the move to Autoconf. I'll remove the paragraph suggesting to edit `Makefile.config`.